### PR TITLE
feat(wallet): Add Loading Spinner To Send Input

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Adds a loading spinner to the input field while searching for `ens` or `ud` domains.

Resolves <https://github.com/brave/brave-browser/issues/27520>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to brave://wallet/send
2. Select a `Token` and `Amount`
3. Enter an `ENS` or `UD` domain
4. There should be a loading spinner in the input until the domain is resolved.

https://user-images.githubusercontent.com/40611140/209169609-a7446744-d52b-473c-846b-044def706811.mov

### CODE REVIEW

**Review by josheleonard — APPROVED** [review]
++

File: components/brave_wallet_ui/page/screens/send/send/send.tsx [hunk 1]
```diff
@@ -242,6 +244,14 @@ export const Send = (props: Props) => {
     return false
   }, [addressError, toAddress, toAddressOrUrl])
 
+  const showSearchingForAddressIcon = React.useMemo(() => {
+    if ((endsWithAny(allSupportedExtensions, toAddressOrUrl.toLowerCase()) && searchingForDomain) || showEnsOffchainLookupOptions) {
+      setSearchingForAddressIconPosition(document.getElementById(INPUT_WIDTH_ID)?.clientWidth ?? 0)
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/send/send/send.tsx:249]: side effects belong in useEffect, not useMemo

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/page/screens/send/send/send.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/send/send/send.tsx]: Removed this and moved to `useEffect` hook.

File: components/brave_wallet_ui/page/screens/send/send/send.tsx [hunk 2]
```diff
@@ -275,6 +265,43 @@ export const Send = (props: Props) => {
...
+      setFoundAddressPosition(document.getElementById(INPUT_WIDTH_ID)?.clientWidth ?? 0)
+      return
+    }
+    // Do nothing if the foundAddressPosition position is already 0
+    if (foundAddressPosition === 0) {
+      return
+    }
+    // Fallback to 0
+    setFoundAddressPosition(0)
+  }, [addressError, toAddress, toAddressOrUrl, foundAddressPosition])
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/send/send/send.tsx:303]:
```suggestion
 const foundAddressPosition = React.useMemo(() => {
    // Sets the foundAddressPosition only if it meets conditions.
    if (
      (addressError === undefined ||
        addressError === '' ||
        addressError === getLocale('braveWalletSameAddressError')) &&
      toAddress &&
      endsWithAny(allSupportedExtensions, toAddressOrUrl.toLowerCase())
    ) {
      return document.getElementById(INPUT_WIDTH_ID)?.clientWidth ?? 0
    }
    // Fallback to 0
    return 0
  }, [addressError, toAddress, toAddressOrUrl, foundAddressPosition])
```

File: components/brave_wallet_ui/page/screens/send/send/send.tsx [hunk 3]
```diff
@@ -275,6 +265,43 @@ export const Send = (props: Props) => {
...
+      setSearchingForAddressIconPosition(document.getElementById(INPUT_WIDTH_ID)?.clientWidth ?? 0)
+      return
+    }
+    // Do nothing if the searchingForAddressIconPosition position is already 0
+    if (searchingForAddressIconPosition === 0) {
+      return
+    }
+    // Fallback to 0
+    setSearchingForAddressIconPosition(0)
+  }, [searchingForDomain, toAddressOrUrl, showEnsOffchainLookupOptions, searchingForAddressIconPosition])
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/send/send/send.tsx:283]:
```suggestion
    const searchingForAddressIconPosition = React.useMemo(() => {
    // Sets the searchingForAddressIconPosition only if it meets conditions.
    if (
      (endsWithAny(allSupportedExtensions, toAddressOrUrl.toLowerCase()) && searchingForDomain) ||
      showEnsOffchainLookupOptions
    ) {
      return document.getElementById(INPUT_WIDTH_ID)?.clientWidth ?? 0
    }
    // Fallback to 0
    return 0
  }, [searchingForDomain, toAddressOrUrl, showEnsOffchainLookupOptions])
```

**Review by josheleonard — CHANGES_REQUESTED** [review]
No extra state or effects needed

File: components/brave_wallet_ui/page/screens/send/send/send.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/send/send/send.tsx]: Updated!

**Review by josheleonard — APPROVED** [review]
++

**Pull request merged**